src/components/UpvoteArticle.jsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react";
+import { patchVotes , patchVotesDown} from "../../utils/api";
+
+
+function UpvoteArticle({currVotes, setCurrVotes, article_id}){
+    const [hasUpVoted, setHasUpVoted]=useState(false)
+    const [hasDownVoted, setHasDownVoted]=useState(false)
```

Comment:
to deal with both buttons staying as disabled when you've clicked on them, instead of setting this state to true you could set it to the opposite value of what is already set too.